packages/react-components/react-combobox/src/components/Option/useOption.tsx

```diff
@@ -67,16 +88,16 @@ export const useOption_unstable = (props: OptionProps, ref: React.Ref<HTMLElemen
       return;
     }
 
-    onOptionClick(event, { id, disabled, value: optionValue });
+    onOptionClick(event);
     props.onClick?.(event);
   };
 
   // register option data with context
   React.useEffect(() => {
     if (id && optionRef.current) {
-      return registerOption({ id, disabled, value: optionValue }, optionRef.current);
+      return registerOption(optionData, optionRef.current);
     }
-  }, [registerOption, id, disabled, optionValue]);
+  }, [id, optionData, registerOption]);
```

Comment:
Just to double check: this is now calling `registerOption` potentially more frequently, whenever optionData or registerOption change. Does `registerOption` handle that case? And/or does the effect need a cleanup function that de-registers it?

Reply:
it should be calling `registerOption` the same amount, since it ends up having the same deps either way. `optionData` should only be redefined when one of the values is changed, since it uses `useMemo`.